robot/impl/resource_manager_test.go

```diff
@@ -1850,3 +1852,72 @@ func managerForDummyRobot(t *testing.T, robot robot.Robot) *resourceManager {
 	}
 	return manager
 }
+
+// TestReconfigureParity validates that calling the synchronous and asynchronous version
+// of `Reconfigure` results in the same resource state on a local robot for all
+// combinations of an initial and updated configuration.
+func TestReconfigureParity(t *testing.T) {
+	files := []string{
+		"data/diff_config_deps1.json",
+		"data/diff_config_deps2.json",
+		"data/diff_config_deps3.json",
+		"data/diff_config_deps4.json",
+		"data/diff_config_deps5.json",
+		"data/diff_config_deps6.json",
+		"data/diff_config_deps7.json",
+		"data/diff_config_deps8.json",
+		"data/diff_config_deps9_good.json",
+		"data/diff_config_deps9_bad.json",
+		"data/diff_config_deps10.json",
+		"data/diff_config_deps11.json",
+		"data/diff_config_deps12.json",
+	}
+	logger := logging.NewTestLogger(t)
+	ctx := context.Background()
+
+	testReconfigureParity := func(t *testing.T, initCfg, updateCfg string) {
+		name := fmt.Sprintf("%s -> %s", initCfg, updateCfg)
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			// Configuration may mutate `*config.Config`, so we read it from
+			// file each time.
```

Comment:
I'm going to drop this test, but separately we should follow-up make it so `*config.Config` doesn't get mutated if possible. This can lead to some pretty confusing test failures.